src/App.tsx

```diff
@@ -1,61 +1,130 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/todo';
+import { User } from './types';
+
+function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId)
+    || null;
+}
+
+function getMaxTodoId(todos: Todo[]) {
+  const todoIds = todos.map(({ id }) => id);
+
+  return Math.max(...todoIds);
+}
+
+export const createTodos = () => (todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+})));
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(createTodos());
+  const [touched, setTouched] = useState({
+    title: false,
+    user: false,
+  });
+  const [newTitle, setNewTitle] = useState('');
+  const [newUserId, setnewUserId] = useState(0);
+
+  const onAdd = (newTodo: Todo) => {
+    setTodos(prevTodos => [...prevTodos, newTodo]);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (!newTitle && newUserId) {
+      setTouched({ title: true, user: false });
+
+      return;
+    }
+
+    if (newTitle && !newUserId) {
+      setTouched({ title: false, user: true });
+
+      return;
+    }
+
+    if (!newTitle && !newUserId) {
+      setTouched({ title: true, user: true });
+
+      return;
+    }
```

Comment:
You can combine this like:
```suggestion
    if (!newTitle || !newUserId) {
      setTouched([set correct value here]);

      return;
    }
```